kola/tests/ignition/security.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2018 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ignition
+
+import (
+	"crypto/tls"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+
+	"github.com/vincent-petithory/dataurl"
+
+	"github.com/coreos/mantle/kola/cluster"
+	"github.com/coreos/mantle/kola/register"
+	"github.com/coreos/mantle/platform/conf"
+	"github.com/coreos/mantle/platform/machine/packet"
+)
+
+var (
+	localSecurityClient = conf.Ignition(`{
+         "ignition": {
+             "version": "2.2.0",
+             "config": {
+               "append": [{
+				   "source": "https://$IP"
+               }]
+             },
+             "security": {
+                 "tls": {
+                     "certificateAuthorities": [{
+                         "source": "$KEY"
+                     }]
+                 }
+             }
+         }
+     }`)
+)
+
+func init() {
+	register.Register(&register.Test{
+		Name:        "coreos.ignition.v2_2.security.tls",
+		Run:         securityTLS,
+		ClusterSize: 1,
+		NativeFuncs: map[string]func() error{
+			"TLSServe": TLSServe,
+		},
+		// https://github.com/coreos/bugs/issues/2205
+		ExcludePlatforms: []string{"do"},
+	})
+}
+
+func securityTLS(c cluster.TestCluster) {
+	server := c.Machines()[0]
+
+	ip := server.PrivateIP()
+	if c.Platform() == packet.Platform {
+		// private IP not configured in the initramfs
+		ip = server.IP()
+	}
+
+	c.MustSSH(server, "openssl ecparam -genkey -name secp384r1 -out server.key")
+	c.MustSSH(server, strings.Replace(`openssl req -new -x509 -sha256 -key server.key -out server.crt -days 3650 -subj "/CN=$IP" -config <(cat <<-EOF
+[req]
+default_bits = 2048
+default_md = sha256
+distinguished_name = dn
+
+[ dn ]
+CN = 192.168.0.20
+
+[ SAN ]
+subjectAltName = IP:$IP
+EOF
+) -extensions SAN`, "$IP", ip, -1))
```

Comment:
The current method looks okay, but if you'd like to do something like write the OpenSSL config file with Ignition, you could try using `${ENV::IP}` in the file to substitute the value with an environment variable.